### ISSUE TITLE
.NET 8 in-process model Could not load file or assembly 'Microsoft.Extensions.DependencyModel, Version=8.0.0.1

### ISSUE THREAD

**Issue by vladyslav-panasenko** [Azure/azure-functions-host#10660]
**Version**
AzureFuntionTools 4.91.0-inprocess
Windows 11
.NET 8
Azure function V4 in-process model

**Description**
Error during Azure function startup

System.IO.FileNotFoundException: 'Could not load file or assembly 'Microsoft.Extensions.DependencyModel, Version=8.0.0.1, Culture=neutral, PublicKeyToken=adb9793829ddae60'. The system cannot find the file specified.'

Serilog.AspNetCore 8.0.2 depends on Microsoft.Extensions.DependencyModel 8.0.1

**Steps to reproduce**
Run Azure function locally
[FunctionApp1.zip](https://github.com/user-attachments/files/17308696/FunctionApp1.zip)

Startup.cs

```
internal class Startup : FunctionsStartup
{
    public IConfiguration Configuration { get; set; }

    public override void ConfigureAppConfiguration(IFunctionsConfigurationBuilder builder)
    {
        var fileInfo = new FileInfo(Assembly.GetExecutingAssembly().Location);
        var path = fileInfo.Directory.Parent.FullName;
        var environment = Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT")?.ToLower() ?? "dev";

        Configuration = builder.ConfigurationBuilder
            .SetBasePath(path)
            .AddJsonFile("appsettings.json", optional: false, reloadOnChange: true)
            .AddJsonFile($"appsettings.{environment}.json", optional: true, reloadOnChange: true)
            .AddJsonFile("local.settings.json", optional: true, reloadOnChange: true)
            .AddEnvironmentVariables()
            .Build();
    }

    public override void Configure(IFunctionsHostBuilder builder)
    {
        ConfigureServices(builder.Services);
    }

    private void ConfigureServices(IServiceCollection services)
    {
        var configReaderOptions = new ConfigurationReaderOptions(ConfigurationAssemblySource.AlwaysScanDllFiles)
        {
            SectionName = ConfigurationLoggerConfigurationExtensions.DefaultSectionName,
        };

        var logger = new LoggerConfiguration()
           .ReadFrom.Configuration(Configuration, configReaderOptions)
           .Enrich.FromLogContext()
           .Enrich.WithCorrelationId()
           .Enrich.WithExceptionDetails()
           .CreateLogger();

        services.AddLogging(lb => lb.AddSerilog(logger));
    }
```

csproj
```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <AzureFunctionsVersion>v4</AzureFunctionsVersion>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="4.4.1" />
    <PackageReference Include="Microsoft.Azure.Functions.Extensions" Version="1.1.0" />
    <PackageReference Include="Serilog.AspNetCore" Version="8.0.2" />
    <PackageReference Include="Serilog.Enrichers.CorrelationId" Version="3.0.1" />
    <PackageReference Include="Serilog.Exceptions" Version="8.4.0" />
  </ItemGroup>
  <ItemGroup>
    <None Update="host.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </None>
    <None Update="appsettings.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
      <CopyToPublishDirectory>Never</CopyToPublishDirectory>
    </None>
    <None Update="local.settings.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
      <CopyToPublishDirectory>Never</CopyToPublishDirectory>
    </None>
  </ItemGroup>
</Project>
```

Simple HTTP trigger
```
    public class Function1
    {
        private readonly ILogger<Function1> _logger;
        public Function1(ILogger<Function1> logger)
        {
            _logger = logger;
        }

        [FunctionName("Function1")]
        public async Task<IActionResult> RunAsync(
            [HttpTrigger(AuthorizationLevel.Function, "get", "post", Route = null)] HttpRequest req)
        {
            _logger.LogInformation("C# HTTP trigger function processed a request.");

            var requestBody = await new StreamReader(req.Body).ReadToEndAsync();

            _logger.LogInformation(requestBody);
            
            return new OkObjectResult(requestBody);
        }
    }
```

**Comment by vladyslav-panasenko** [comment]
I tried using the latest Azure functions tools (4.97.0) and Microsoft.NET.Sdk.Functions 4.5.0. Same issue. Any suggestions?

**Comment by bhagyshricompany** [comment]
Hi @vladyslav-panasenko  Thanks for updating will check  and update you asap.Thanks

**Comment by bhagyshricompany** [comment]
Hi @jviau please comment Thanks

**Comment by jviau** [comment]
Thank you for reporting this. I have identified we were not pinning this assembly and have put up a PR to address that

**Comment by vladyslav-panasenko** [comment]
Hello @bhagyshricompany  @jviau. Do you know when/how the fix will be released?

**Comment by jviau** [comment]
It did not make it into the release we are processing now, so it will be in the release following that (4.38.0). Don't have an ETA, but it will be at least March 2025.

**Comment by jviau** [comment]
Resolved by #10661
